tools/github/before_install.sh

```diff
@@ -29,6 +29,10 @@ if [[ $MINIMUM_REQUIREMENTS == 1 ]]; then
     done
 fi
 
+
+perl -pi -e 's/numpy>=1.21.1/numpy>=1.21.1,<1.25/g' requirements/default.txt
+perl -pi -e 's/numpy>=1.21.1/numpy>=1.21.1,<1.25/g' requirements/build.txt
```

Comment:
I think this needs to be removed?

Reply:
Those lines prevent disables testing on NumPy 1.25. That is the point of 74d533c.

Without that the tests will continue to fail on main and on all PRs from main. The idea is to merge this PR and make a PR to remove those lines and figure out how to fix the remaining errors with NumPy 1.25.